src/app/clusters/thermostat-server/thermostat-server-suggestions.cpp

```diff
@@ -0,0 +1,186 @@
+
+    // If the effective time in UTC is greater than current time in UTC plus 24 hours, return INVALID_COMMAND.
+    const uint32_t kSecondsInDay = 24 * 60 * 60;
+    if (!commandData.effectiveTime.IsNull() &&
+        (commandData.effectiveTime.Value() > currentMatterEpochTimestampInSeconds + kSecondsInDay))
+    {
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return true;
+    }
+
+    uint8_t uniqueID = delegate->GetUniqueID();
+
+    Structs::ThermostatSuggestionStruct::Type thermostatSuggestion;
+    thermostatSuggestion.uniqueID     = uniqueID;
+    thermostatSuggestion.presetHandle = commandData.presetHandle;
+
+    uint32_t effectiveTime             = commandData.effectiveTime.ValueOr(currentMatterEpochTimestampInSeconds);
+    thermostatSuggestion.effectiveTime = effectiveTime;
+
+    const uint32_t kSecondsInMinute     = 60;
+    thermostatSuggestion.expirationTime = effectiveTime + (commandData.expirationInMinutes * kSecondsInMinute);
+
+    CHIP_ERROR err = delegate->AppendToThermostatSuggestionsList(thermostatSuggestion);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to AppendToThermostatSuggestionsList with error: %" CHIP_ERROR_FORMAT, err.Format());
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return true;
+    }
+
+    MatterReportingAttributeChangeCallback(endpoint, Thermostat::Id, ThermostatSuggestions::Id);
+
+    err = delegate->ReEvaluateCurrentSuggestion();
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to ReEvaluateCurrentSuggestion with error: %" CHIP_ERROR_FORMAT, err.Format());
+    }
+
+    Commands::AddThermostatSuggestionResponse::Type response;
+    response.uniqueID = uniqueID;
+    commandObj->AddResponse(commandPath, response);
+    return true;
+}
+
+bool RemoveThermostatSuggestion(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                const Commands::RemoveThermostatSuggestion::DecodableType & commandData)
+{
+    EndpointId endpoint = commandPath.mEndpointId;
+    auto delegate       = GetDelegate(endpoint);
+    if (delegate == nullptr)
+    {
+        ChipLogError(Zcl, "Delegate is null");
+        commandObj->AddStatus(commandPath, Status::InvalidInState);
+        return true;
+    }
+
+    delegate->SetEndpointId(endpoint);
+
+    CHIP_ERROR err = delegate->RemoveFromThermostatSuggestionsList(commandData.uniqueID);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to RemoveFromThermostatSuggestionsList with uniqueID: %d error: %" CHIP_ERROR_FORMAT,
+                     commandData.uniqueID, err.Format());
+        commandObj->AddStatus(commandPath, Status::NotFound);
+        return true;
+    }
+
+    MatterReportingAttributeChangeCallback(endpoint, Thermostat::Id, ThermostatSuggestions::Id);
+    commandObj->AddStatus(commandPath, Status::Success);
+
+    uint32_t currentMatterEpochTimestampInSeconds = 0;
```

Comment:
Why is this here?  We get this value and then... do nothing with it.  What's the point?

Reply:
forgot to remove.

Reply:
removed.